### PR TITLE
Changed the deprecated usage of torch.Tensor.grad.data

### LINKED ISSUE #3217: torch.Tensor.grad.data attribute is deprecated - update it's usage
PyTorch is deprecating `torch.Tensor.grad.data`. We are using `data` attribute at some parts of the code, in torch models. The usage should be updated, maybe with `tensor.detach()`?

### PR DESCRIPTION
**Pull request by HridayM25**
## Description

Fix #3217

## Type of change

Please check the option that is related to your PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by rbharath — COMMENTED** [review]
@shaipranesh2 Can you review this PR?

**Review by shaipranesh2 — CHANGES_REQUESTED** [review]
@HridayM25 The code formatting test are failing due to change in yapf version

**Comment by HridayM25** [comment]
@shaipranesh2 Thanks for the review! I am using yapf whose version is 0.32.0. How should I proceed?

**Comment by shaipranesh2** [comment]
@HridayM25 Rest seems fine as detach() here does not alter the computation graph of the KFAC. Should be ready to merge once code formatting tests pass

**Comment by shaipranesh2** [comment]
> @shaipranesh2 Thanks for the review! I am using yapf whose version is 0.32.0. How should I proceed?

I am not sure then. The CI tests expected a different formatting than the one in this PR. So check the logs and rerun the yapf command maybe.

**Comment by HridayM25** [comment]
@shaipranesh2 I checked the logs, and I have made the necessary changes. I ran yapf and it gave me no errors.

**Comment by HridayM25** [comment]
@rbharath @shaipranesh2 The formatting issue seems to be solved now.

**Review by rbharath — COMMENTED** [review]
LGTM

**Review by rbharath — APPROVED** [review]
LGTM
